src/init/device.rs

```diff
@@ -301,15 +285,22 @@ impl QueueRequest {
 }
 
 impl VulkanQueue {
+    pub fn access_queue(&self) -> &Mutex<Queue> {
+        &self.queue
+    }
+
     pub fn queue_submit(&self, device: ash::Device, submits: &[SubmitInfo], fence: Fence) -> VkResult<()> {
-        unsafe { device.queue_submit(self.queue, submits, fence) }
+        let guard = self.queue.lock().unwrap();
```

Comment:
I feel like this could be propagated up instead of panicing, but it's probably not a big deal

Reply:
Agreed i want to do that but due to time constraints i opted to just panic. If we want to propagate we would need to define a new error type that includes poisoned mutexs and vk errors. This is also something i needed all over the place in the execution engine. I wonder how granular we should make our error types. There's a decent discussion to be had here i feel.